### ISSUE TITLE
can you help me understand currMask in the getDatasetContinuous.py script?

### ISSUE THREAD

**Issue by yudeng2022** [btheodorou99/HALO_Inpatient#12]
Hi 
Thank you so much for providing the open source code for your HALO paper. 
I am reading through the getDatasetContinuous.py script and trying to generate one for my own data. I have trouble understanding the following piece of code. 
```
for col, value in row.items(): #changed iteritems to items();#iteritems is removed from pandas 2.0
                if value != value or col == 'Hours':
                    continue
                #print(f"=================col is {col}, val is {value}========================")
                if is_categorical_channel[col]:
                    if col == 'Glascow coma scale total':
                        value = str(int(value))
                    elif col == 'Capillary refill rate':
                        value = str(value)
                    if begin_pos[channel_to_id[col]] in currMask:
                        currValues[currMask.index(begin_pos[channel_to_id[col]] + possible_values[col].index(value))] = 1
                    else:
                        for j in range(begin_pos[channel_to_id[col]], end_pos[channel_to_id[col]]):
                            currMask.append(j)
                            currValues.append(1 if j - begin_pos[channel_to_id[col]] == possible_values[col].index(value) else 0)
                else:
                    if begin_pos[channel_to_id[col]] in currMask:
                        currValues[currMask.index(begin_pos[channel_to_id[col]])] = value
                    else:
                        currMask.append(begin_pos[channel_to_id[col]])
                        currValues.append(value)

```

Could you help me understand what currMask does by any chance and what begin_pos, end_pos files are for?

**Comment by btheodorou** [comment]
Hello! To be honest the approach is probably not the best (and in subsequent experiments, we have moved away from it), but I will outline what the code you mentioned does. currMask basically contains the lab index (where lab indices include multiple slots for categorical labs and a single slot for continuous labs). Begin_pos then maps an overall lab number (where each of categorical and continuous labs have a single spot) to the spot where that lab's indices start. End pos is where that lab's indices end (and if it's continuous it's only one so you don't need an end_pos). However, a lot of this gets overwritten in discretize.py which then discretizes the continuous labs so that they don't all have just a single value. The code is messy and the masks themselves are an artifact of previous experiments where we modeled the presence of a lab separate from its values. However, in general this separation of currMask and currValues as well as separation of this value curation from discretization just makes things messier and harder to understand. We have since moved to discretizing out front (by collecting all the values for each lab) and then just operating over variables here in the same way as everywhere else where a lab value pair just becomes a new medical code basically (though this code from the paper experiments is still the old messy way).
